File: src/main/java/com/godwin/jsonparser/ui/action/OpenParserAction.java
package com.godwin.jsonparser.ui.action;

import com.godwin.jsonparser.ParserToolWindowFactory;
import com.godwin.jsonparser.common.Logger;
import com.godwin.jsonparser.rx.Subscriber;
import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.editor.SelectionModel;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowManager;
import org.apache.http.util.TextUtils;
import org.jetbrains.annotations.NotNull;

import static com.godwin.jsonparser.constants.ConstantsKt.TOOL_WINDOW_ID;

public class OpenParserAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
        try {
            Project project = anActionEvent.getProject();
            Editor mEditor = anActionEvent.getData(PlatformDataKeys.EDITOR);
            if (null == mEditor) {
                Logger.i("Not in editor window");
                return;
            }

            SelectionModel model = mEditor.getSelectionModel();
            String selectText = model.getSelectedText();

            if (TextUtils.isEmpty(selectText)) {
                Logger.i("selectText == null");
                return;
            }

            if (project != null) {
                ToolWindow window = ToolWindowManager.getInstance(project).getToolWindow(TOOL_WINDOW_ID);
                if (window != null) {
                    window.activate(() -> {
                        Subscriber.INSTANCE.publishMessage(selectText);
                    }, true, true);
                }
            }
        } catch (Exception e) {
            Logger.e(e.getMessage());
        }
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
    }

    @Override
    public @NotNull ActionUpdateThread getActionUpdateThread() {
        return ActionUpdateThread.BGT;
    }
}

File: src/main/kotlin/com/godwin/jsonparser/action/OpenParserAction.kt
package com.godwin.jsonparser.action

import com.godwin.jsonparser.constants.TOOL_WINDOW_ID
import com.godwin.jsonparser.rx.Subscriber
import com.godwin.jsonparser.util.Log
import com.godwin.jsonparser.util.analytics.AnalyticsConstant
import com.godwin.jsonparser.util.analytics.AnalyticsService
import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.fileEditor.FileEditorManager
import com.intellij.openapi.wm.ToolWindowManager

class OpenParserAction : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        AnalyticsService.track(AnalyticsConstant.ACTION_OPEN_PARSER)
        try {
            val project = e.project ?: return
            val editor = FileEditorManager.getInstance(project).selectedTextEditor
            if (editor == null) {
                Log.i("Not in editor window")
                return
            }

            val selectText = editor.selectionModel.selectedText
            if (selectText.isNullOrEmpty()) {
                Log.i("selectText == null : ${selectText == null}")
                return
            }

            ToolWindowManager.getInstance(project).getToolWindow(TOOL_WINDOW_ID)?.activate({
                Subscriber.publishMessage(selectText)
            }, true, true)
        } catch (e: Exception) {
            Log.e(e.message)
        }
    }

    override fun getActionUpdateThread() = ActionUpdateThread.BGT
}
